actors/builtin/multisig/multisig_test.go

```diff
@@ -75,10 +78,214 @@ func TestConstruction(t *testing.T) {
+			Method:   0,
+			Params:   abi.MethodParams{},
+			Approved: []addr.Address{anne},
+		}, txn)
+	})
+
+	t.Run("propose with threshold met", func(t *testing.T) {
+		rt := builder.WithBalance(abi.NewTokenAmount(20), abi.NewTokenAmount(0)).Build()
+		constructParams := multisig.ConstructorParams{
+			Signers:               []addr.Address{anne, bob},
+			NumApprovalsThreshold: 1,
+			UnlockDuration:        0,
+		}
+
+		rt.ExpectValidateCallerAddr(builtin.InitActorAddr)
+		constructRet := actor.Constructor(rt, &constructParams)
+		assert.Equal(t, runtime.EmptyReturn{}, *constructRet)
+		rt.Verify()
+
+		// going to assume the state is correct, a method that constructs and verifies would be nice.
+		proposeParams := &multisig.ProposeParams{
+			To:     chuck,
+			Value:  abi.NewTokenAmount(10),
+			Method: 0,
+			Params: nil,
+		}
+
+		rt.ExpectSend(chuck, 0, nil, abi.NewTokenAmount(10), nil, 0)
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.Propose(rt, proposeParams)
+		rt.Verify()
+
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		txns := adt.AsMap(rt.Store(), st.PendingTxns)
+		keys, err := txns.CollectKeys()
+		require.NoError(t, err)
+		assert.Empty(t, keys)
+	})
+
+	t.Run("fail propose with threshold met and insufficient balance", func(t *testing.T) {
+		rt := builder.WithBalance(abi.NewTokenAmount(0), abi.NewTokenAmount(0)).Build()
+		constructParams := multisig.ConstructorParams{
+			Signers:               []addr.Address{anne, bob},
+			NumApprovalsThreshold: 1,
+			UnlockDuration:        0,
+		}
+
+		rt.ExpectValidateCallerAddr(builtin.InitActorAddr)
+		constructRet := actor.Constructor(rt, &constructParams)
+		assert.Equal(t, runtime.EmptyReturn{}, *constructRet)
+		rt.Verify()
+
+		// going to assume the state is correct, a method that constructs and verifies would be nice.
+		proposeParams := &multisig.ProposeParams{
+			To:     chuck,
+			Value:  abi.NewTokenAmount(10),
+			Method: 0,
+			Params: nil,
+		}
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectAbort(exitcode.ErrInsufficientFunds, func() {
+			actor.Propose(rt, proposeParams)
+		})
+		rt.Verify()
+
+		// even though propose aborts there should still be a transaction.
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		txns := adt.AsMap(rt.Store(), st.PendingTxns)
+		keys, err := txns.CollectKeys()
+		require.NoError(t, err)
+		// there is exactly one transaction
+		assert.Equal(t, 1, len(keys))
+
+		var txn multisig.MultiSigTransaction
+		found, err := txns.Get(keys[0], &txn)
+		assert.True(t, found)
+
+		// containing these values
+		assert.Equal(t, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(10),
+			Method:   0,
+			Params:   abi.MethodParams{},
+			Approved: []addr.Address{anne},
+		}, txn)
```

Comment:
This doesn't look right, in the event that the multisig actor has insufficient funds the transaction is still kept in its state.